components/script/bluetooth_blacklist.rs

```diff
@@ -0,0 +1,98 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use regex::Regex;
+use std::cell::RefCell;
+use std::collections::HashMap;
+use std::fs::File;
+use std::io::Read;
+use std::path::Path;
+use std::string::String;
+
+const EXCLUDE_READS: &'static str = "exclude-reads";
+const EXCLUDE_WRITES: &'static str = "exclude-writes";
+const FILE_PATH: &'static str = "./resources/gatt_blacklist.txt";
+const VALID_UUID_REGEX: &'static str = "^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}";
+
+thread_local!(pub static BLUETOOTH_BLACKLIST: RefCell<Option<BluetoothBlacklist>> =
+              RefCell::new(BluetoothBlacklist::init()));
+
+pub struct BluetoothBlacklist(HashMap<String, ValueType>);
+
+#[derive(Eq, PartialEq)]
+pub enum ValueType {
+    Exclude,
+    ExcludeReads,
+    ExcludeWrites,
+}
+
+impl BluetoothBlacklist {
+    pub fn init() -> Option<BluetoothBlacklist> {
```

Comment:
It's easier to check the None, than the emptiness of the HashMap field in my opinion. And if we want to check the emptiness of the HashMap, we should make it public or create a getter.